sycl/test-e2e/Basic/kernel_info.cpp

```diff
@@ -82,12 +89,70 @@ int main() {
+  assert(prefWGSizeMultExt == prefWGSizeMult);
+  const cl_uint maxSgSizeExt = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::max_sub_group_size>(ctx, dev);
+  assert(maxSgSizeExt == maxSgSize);
+  const cl_uint compileSgSizeExt = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::compile_sub_group_size>(ctx,
+                                                                        dev);
+  assert(compileSgSizeExt == compileSgSize);
+  const cl_uint maxNumSgExt = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::max_num_sub_groups>(ctx, dev);
+  assert(maxNumSgExt == maxNumSg);
+  const cl_uint compileNumSgExt = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::compile_num_sub_groups>(ctx,
+                                                                        dev);
+  assert(compileNumSgExt == compileNumSg);
+
+  // Use ext_oneapi_get_kernel_info extension with queue parameter and check the
+  // result.
+  const size_t wgSizeExtQ =
+      syclex::get_kernel_info<SingleTask,
+                              info::kernel_device_specific::work_group_size>(q);
+  assert(wgSizeExtQ == wgSize);
+  const size_t prefWGSizeMultExtQ = syclex::get_kernel_info<
+      SingleTask,
+      info::kernel_device_specific::preferred_work_group_size_multiple>(q);
+  assert(prefWGSizeMultExtQ == prefWGSizeMult);
+  const cl_uint maxSgSizeExtQ = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::max_sub_group_size>(q);
+  assert(maxSgSizeExtQ == maxSgSize);
+  const cl_uint compileSgSizeExtQ = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::compile_sub_group_size>(q);
+  assert(compileSgSizeExtQ == compileSgSize);
+  const cl_uint maxNumSgExtQ = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::max_num_sub_groups>(q);
+  assert(maxNumSgExtQ == maxNumSg);
+  const cl_uint compileNumSgExtQ = syclex::get_kernel_info<
+      SingleTask, info::kernel_device_specific::compile_num_sub_groups>(q);
+  assert(compileNumSgExtQ == compileNumSg);
+
   {
     std::error_code Errc;
     std::string ErrMsg = "";
     bool IsExceptionThrown = false;
     try {
-      krn.get_info<sycl::info::kernel_device_specific::global_work_size>(dev);
+      auto globalWorkSize =
+          krn.get_info<sycl::info::kernel_device_specific::global_work_size>(
+              dev);
+      // Use ext_oneapi_get_kernel_info extension and check that answers match.
+      auto globalWorkSizeExt = syclex::get_kernel_info<
+          SingleTask, info::kernel_device_specific::global_work_size>(ctx, dev);
+      assert(globalWorkSize == globalWorkSizeExt);
+      // Use ext_oneapi_get_kernel_info extension with queue parameter and check
+      // the result.
+      auto globalWorkSizeExtQ = syclex::get_kernel_info<
+          SingleTask, info::kernel_device_specific::global_work_size>(q);
+      assert(globalWorkSize == globalWorkSizeExtQ);
```

Comment:
Same comment here about exceptions.  The call to `krn.get_info<sycl::info::kernel_device_specific::global_work_size>()` throws an exception, so this test never even makes the two calls to `syclex::get_kernel_info`.

Reply:
Fixed.